onnx_xla/operator_registry.cc

```diff
@@ -41,6 +43,36 @@ namespace onnx_xla  {
     return ONNXIFI_STATUS_SUCCESS;
   }
   REGISTER_OPERATOR_TRANSLATOR(Relu, translateRelu)
+
+  //To be tested
+  //If 1 input, match output value to same op input value is matched to
+  //Otherwise do successive additions, being sure to satisfy broadcast semantics
+    //If same rank or scalar, no need to specify
+    //otherwise, use numpy matching of trailing dimensions
+  onnxStatus translateSum(const Node& n, XlaBuilder& builder, ValueOpMap& valueToOp)  {
+    if (n.inputs().size() == 1)  {
+      valueToOp[n.outputs().at(0)] = valueToOp[n.inputs().at(0)];
+      return ONNXIFI_STATUS_SUCCESS;
+    }
+    auto firstOp = valueToOp[n.inputs().at(0)];
+    for (auto i = 1; i < n.inputs().size(); ++i)  {
+      auto secondOp = valueToOp[n.inputs().at(i)];
+      auto firstNDim = ShapeUtil::Rank(builder.GetShape(firstOp).ValueOrDie());
+      auto secondNDim = ShapeUtil::Rank(builder.GetShape(secondOp).ValueOrDie());
+      std::vector<int64> broadcastDims = {};
+      if (firstNDim != secondNDim || firstNDim != 0 || secondNDim != 0)  {
+        auto minDim = std::min(firstNDim, secondNDim);
+        auto maxDim = std::max(firstNDim, secondNDim);
+        for (auto j = 0; j < minDim; ++j)  {
+          broadcastDims.push_back(j + maxDim - minDim);
+        }
+      }
+      firstOp = builder.Add(firstOp, secondOp, broadcastDims);
```

Comment:
So XLA builder doesn't actually have an sum op that supports k-nary add?

Reply:
Unfortunately not, but i believe sum (at least in resnet) are over only a few operands